workflow-runtime/src/test/java/com/squareup/workflow1/internal/ChainedWorkflowInterceptorTest.kt

```diff
@@ -173,44 +171,102 @@ class ChainedWorkflowInterceptorTest {
       override fun <P, S, O, R> onRender(
         renderProps: P,
         renderState: S,
-        context: BaseRenderContext<P, S, O>,
-        proceed: (P, S, BaseRenderContext<P, S, O>) -> R,
+        proceed: (P, S, RenderContextInterceptor<P, S, O>?) -> R,
         session: WorkflowSession
-      ): R = ("r1: " +
-          proceed(
-              "props1: $renderProps" as P,
-              "state1: $renderState" as S,
-              FakeRenderContext("context1: $context") as BaseRenderContext<P, S, O>
-          )) as R
+      ) = "r1: ${proceed("props1: $renderProps" as P, "state1: $renderState" as S, null)}" as R
     }
     val interceptor2 = object : WorkflowInterceptor {
       override fun <P, S, O, R> onRender(
         renderProps: P,
         renderState: S,
-        context: BaseRenderContext<P, S, O>,
-        proceed: (P, S, BaseRenderContext<P, S, O>) -> R,
+        proceed: (P, S, RenderContextInterceptor<P, S, O>?) -> R,
         session: WorkflowSession
-      ): R = ("r2: " +
-          proceed(
-              "props2: $renderProps" as P,
-              "state2: $renderState" as S,
-              FakeRenderContext("context2: $context") as BaseRenderContext<P, S, O>
-          )) as R
+      ) = "r2: ${proceed("props2: $renderProps" as P, "state2: $renderState" as S, null)}" as R
     }
     val chained = listOf(interceptor1, interceptor2).chained()
-    fun render(
-      props: String,
-      state: String,
-      context: BaseRenderContext<String, String, String>
-    ): String = "($props|$state|$context)"
 
     val finalRendering =
-      chained.onRender("props", "state", FakeRenderContext("context"), ::render, TestSession)
+      chained.onRender<String, String, Nothing, Any>(
+        "props", "state", { p, s, _ -> "($p|$s)" }, TestSession
+      )
 
     assertEquals(
-        "r1: r2: (props2: props1: props|state2: state1: state|context2: context1: context)",
-        finalRendering
+      "r1: r2: (props2: props1: props|state2: state1: state)",
+      finalRendering
+    )
+  }
+
+  @Test fun `chains calls with RenderContextInterceptor in left-to-right order`() {
```

Comment:
@zach-klippenstein PTAL, new test

Reply:
Lg, although I probably would have made that transcript a list of strings instead. This works though, thanks for writing!

Reply:
It is…

```kotlin
    val transcript = mutableListOf<String>()
```